### PR TITLE
Clarify the behavior of SIGNEXTEND

### PR DESCRIPTION
**Pull request by charles-cooper**
I've found explanations of signextend to be confusing, this change tries to clarify. The key point for me was realizing that b corresponds to the number of bytes used to interpret x.

### CODE REVIEW

**Comment by wolflo** [comment]
Thanks for this! I'm hesitant on this for a few reasons:
1. Getting into describing the semantics of opcodes is a bit of a rabbit hole, and I think would make this reference less digestable.
2. Maybe silly, but I want to keep the notes section all on a single short line to contain the size of the gas table, especially for opcodes that are used or referenced less.
3. `SIGNEXTEND` is hard to understand by any short description, and you really need to refer to some two's complement minutia to understand it.

Do you know of a good reference on sign extension we could link to maybe?

**Comment by charles-cooper** [comment]
Sure, we could link to the wiki page on sign extension for example. But even if one understands in general what sign extension is, the meaning of the opcode itself is pretty confusing because it's unclear in plain English what the `b` parameter is for. That's what I was trying to address here.

**Comment by wolflo** [comment]
Do you think it would improve clarity if we just said "sign extend `x` from b+1 bytes to 32 bytes"?

**Comment by charles-cooper** [comment]
Yes! Brilliant. I will make that change later today.

**Comment by wolflo** [comment]
Thank you!
